src/sagemaker/jumpstart/factory/model.py

```diff
@@ -218,6 +216,23 @@ def _add_model_data_to_kwargs(kwargs: JumpStartModelInitKwargs) -> JumpStartMode
         sagemaker_session=kwargs.sagemaker_session,
     )
 
+    if isinstance(model_data, str) and model_data.startswith("s3://") and model_data.endswith("/"):
+        if kwargs.model_data:
+            JUMPSTART_LOGGER.info(
+                "S3 prefix model_data detected for JumpStartModel: '%s'. "
+                "Converting to S3DataSource dictionary.",
```

Comment:
nit: we could output the S3DataSource dictionary here as well for full visibility that customers can map back to the API docs for Hosting